david_recur2.py

```diff
@@ -0,0 +1,19 @@
+# function returning all permutations of chars of a given string in alphabetical order
+
+def find_all_permutations (input_string):   
+    all_permutations = []
+    input_string = "".join(sorted(input_string))
```

Comment:
Why do you need to sort the string? It's a lot of overhead for not affecting the result at all